src/components/PrimaryPanes/index.js

```diff
@@ -96,7 +98,9 @@ class PrimaryPanes extends Component<Props, State> {
 
   renderThreadSources() {
     const threads = sortBy(
-      Object.getOwnPropertyNames(this.props.sources),
+      Object.getOwnPropertyNames(this.props.sources).filter(
+        this.props.isValidThread
+      ),
```

Comment:
if we're removing the sources from the store we shouldn't be worried about rendering them here.

Reply:
yep - will be a nice followup, but could be difficult